main.go

```diff
@@ -117,18 +124,49 @@ func main() { //nolint:funlen
 
 	ctrl.SetLogger(logger.ConfigLoggers(logmode))
 
+	//	var namespaceSelector labels.Set{labels.ODH.OwnedNamespace: "true"}.AsSelector()
+	// CSVFields := fields.Set{"spec.displayName": string(cluster.OpenDataHub)}
+	cacheOptions := cache.Options{
+		Scheme: scheme,
+		ByObject: map[client.Object]cache.ByObject{
+			// all CRD because it is mainly for pipeline v1 and v2
+			&apiextv1.CustomResourceDefinition{}: {},
+			// For dashboard secret, we can remove it when we move to SSO
+			&corev1.Secret{}: {
+				Field: fields.Set{"metadata.name": "dashboard-oauth-config-generated"}.AsSelector(),
+			},
+			// For trustedCA bundle
+			&corev1.ConfigMap{}: {
+				Field: fields.Set{"metadata.name": trustedcabundle.CAConfigMapName}.AsSelector(),
+				Label: pkglables.Set{labels.K8SCommon.PartOf: "opendatahub-operator"}.AsSelector(),
+			},
+			// For catsrc (avoid frequently check cluster type)
+			&ofapiv1alpha1.CatalogSource{}:         {},
+			&ofapiv1alpha1.ClusterServiceVersion{}: {},
+		},
+	}
+
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{ // single pod does not need to have LeaderElection
-		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
+		Scheme:  scheme,
+		Metrics: ctrlmetrics.Options{BindAddress: metricsAddr},
+		WebhookServer: ctrlwebhook.NewServer(ctrlwebhook.Options{
```

Comment:
it cannot return nil so webhook server should be set. But why default does not work?

Reply:
then i dont even need an error return from this SetupWithManager(), which has no error in any internal call